### DISCUSSION TITLE
Explicit ODEs?

### DISCUSSION THREAD

**Discussion by chelseas** [casadi/casadi#3563] [Q&A]
Hello!
I am looking to encode an explicit ODE using casadi, e.g. it has terms of the form `xdot(t) = -5*t` but I  cannot figure out how to do this in the python interface.  I found this page for C++ https://casadi.sourceforge.net/api/html/dc/d18/chapter4.html  but I'm not sure how to translate it.

**Comment by sandeep026** [comment]
If RHS has time explicitly, you need to do a scaling from $t \in [t_0,t_f]$ to $\tau \in [0,1]$.

You can check out the example from this dicussion https://github.com/casadi/casadi/discussions/3549.  
Hope that helps.

**Reply by sandeep026** [reply]
# addendum

Scaling is not neccessary for dealing with explcit odes.
```
import casadi as cs
t=cs.MX.sym('t',1,1)
x=cs.MX.sym('x',1,1)
f=2*t
x0=0
tg=cs.np.linspace(0,10,10)
ode={'t':t,'ode':f,'x':x}
i=cs.integrator('i','cvodes',ode,tg[0],tg[1:],{'reltol':1e-3})
r=i(x0=x0)
print(tg[-1]**2,':',r['xf'][-1])
```
*Scaling will be in particular useful when you don't want to create multiple integrator objects for different t0, tf which can arise while solving trajectoy optimization in a multiple shooting setting. The scaled version can be reused as t0,tf can be now passed as parameters.*
